### ISSUE TITLE
The theme_file_formatter_table() doesn't respect singular/plural forms

### ISSUE THREAD

**Issue by alanmels** [backdrop/backdrop-issues#5877]
## Description of the bug

The theme_file_formatter_table() always outputs a singular form of the title "Attachment" regardless if just one or more files are uploaded.

## Steps To Reproduce

To reproduce the behavior:
1. Go to any content type, add a file field and on the _Manage Display_ tab set _Table of files_ format for the field.

![Screenshot 2022-12-05 at 7 29 40 PM](https://user-images.githubusercontent.com/56604405/205786169-719623b1-fe9f-4568-a660-891ba1196aaa.png)

2. Create a node of the content type, attach one file, submit the node and you will see a nice table rendered with the header title "Attachment" for the single file.

3. Now, edit the same node or create new one this time uploading two or more files and you will see the singular form of the title doesn't change.

## Expected behavior

Would be nice to have the title take singular or plural forms depending on the number of the files uploaded like:

![Screenshot 2022-12-05 at 7 33 25 PM](https://user-images.githubusercontent.com/56604405/205786654-7ed4ea53-95fb-4057-a8b9-57f4662a181f.png)

![Screenshot 2022-12-05 at 7 33 31 PM](https://user-images.githubusercontent.com/56604405/205786666-c178f6a2-0f90-44fa-9fef-6978801eadfe.png)

**Comment by klonos** [comment]
Thanks @alanmels for the suggestion and the PR. @kiamlaluno has provided some feedback in the PR that needs to be addressed.

**Comment by alanmels** [comment]
@klonos, thanks. I've committed your suggestion after seeing the translation takes within the [format_plural](https://docs.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/format_plural/1) function.

**Comment by bugfolder** [comment]
Code reviewed. LGTM.

**Comment by argiepiano** [comment]
This is an interesting take column headers. We usually don't pluralize headers. For example, the list of user account looks like this:

<img width="1206" alt="Screen Shot 2023-09-18 at 3 21 06 PM" src="https://github.com/backdrop/backdrop-issues/assets/9938978/0fd4648c-113a-4c01-a61d-4e061e1c0f26">

So, it's `Username`, not `Usernames`, because each row cell contains one username. On the other hand "Roles" is "Roles" because there can be more than one roll in each cell. 

This is truly minor, but the PR makes things inconsistent with the rest of the way tables are handled in Backdrop and Drupal. I'm fine either way...

**Comment by alanmels** [comment]
@argiepiano, thanks for the input. You are right about the standards, however the example you brought in is mostly for admins to see whereas the table of files are for mostly the end-users to see. I don't care about certain things on admins' UI as long as it works for the Backdrop community, however I do believe whatever the end-users see ideally must follow the language syntax and if there are multiple attachments then display plural form of that word.

**Comment by bugfolder** [comment]
In this particular case, I think it makes sense and improves UX. So maybe "always pluralize headers" is too much (and in places, too difficult), doing it when it's possible and improves the UX seems like a good thing.

**Comment by olafgrabienski** [comment]
> In this particular case, I think it makes sense and improves UX.

Hm, I'm not sure if it's really an improvement in this case. In German, an attachment ("ein Dateianhang") doesn't only represent one file, it can also stand for an amount of several files. While some people use the plural for multiple files ("Dateianhänge"), others don't, and the plural is however not necessary in German. Is this similar in English, or do you normally use the plural?

**Comment by alanmels** [comment]
@olafgrabienski, if you have bunch of files listed on the end-users' interface then the singular form of "Attachment" looks weird in English, and in this kind of cases ideally the title should be spelled in plural.

**Comment by kiamlaluno** [comment]
I am with @argiepiano in this case: It is *Attachment* because each row is for an attachment. Similarly, it is *Size* because each row shows a single size.

**Comment by bugfolder** [comment]
> I am with @argiepiano in this case: It is Attachment because each row is for an attachment. Similarly, it is Size because each row shows a single size.

Good point. "Attachments" (plural) would be appropriate as the label for the overall table, but in the table header, it's saying what is in this column of each row. So "Attachments" would only be appropriate if a single row contained multiple items, as in the "Roles" and "Operations" in @argiepiano's example. (I know you said this, @argiepiano; it only now sunk in to me.)
